### ISSUE TITLE
Generated Method names contains invalid characters. 

### ISSUE THREAD

**Issue by damianh** [christianhelle/refitter#56]
**Describe the bug**
Hello there, interesting project so gave it a go :)

The following command `refitter https://api.hubspot.com/api-catalog-public/v1/apis/events/v3/send --namespace HubSpot --output ./GeneratedCode.cs` generated this code 

```csharp
namespace HubSpot
{
    public interface ICustomBehavioralEventsAPI
    {

        /// <summary>
        /// Endpoint to send an instance of a behavioral event
        /// </summary>
        [Post("/events/v3/send")]
        Task Post/events/v3/send([Body] BehavioralEventHttpCompletionRequest body);

    }


}
```

As you can see the method name is not valid C#.

**Support Key**: `kedfgny`

**OpenAPI Specifications**
https://api.hubspot.com/api-catalog-public/v1/apis/events/v3/send

**Additional context**

**Comment by damianh** [comment]
The issue seems related to this bit of code.

https://github.com/christianhelle/refitter/blob/fdac7faf29d18165f3a037887ee331d2d3ca5714/src/Refitter.Core/RefitInterfaceGenerator.cs#L50-L53

**Comment by damianh** [comment]
I asked ChatGPT lol.

----

You are seeing the path in the operation name because of how NSwag generates operation names by default. NSwag uses the `OperationNameGenerator` to determine how to create operation names. By default, it uses the `DefaultOperationNameGenerator`, which simply concatenates the HTTP method and the path.

If you want to generate a different style of operation name, you will have to provide a custom implementation of `IOperationNameGenerator`. Here's a sample implementation that should create operation names in the format you want:

```csharp
using System.Linq;
using NJsonSchema;
using NSwag;

public class CustomOperationNameGenerator : IOperationNameGenerator
{
    public bool SupportsMultipleClients { get; } = false;

    public string GetClientName(OpenApiDocument document, string path, OpenApiOperation operation, OpenApiSchemaResolver schemaResolver)
    {
        return string.Empty;
    }

    public string GetOperationName(OpenApiDocument document, string path, string httpMethod, OpenApiOperation operation, OpenApiSchemaResolver schemaResolver)
    {
        var operationName = $"{httpMethod.ToUpperInvariant()}{string.Join(string.Empty, path.Split('/').Select(p => p.ConvertKebabCaseToPascalCase()))}";

        return operationName;
    }
}
```

This implementation splits the path by `/`, converts each part from kebab case to Pascal case, then joins them together and prepends the HTTP method.

To use this `OperationNameGenerator`, set it in the `BaseSettings` of your `generator`:

```csharp
generator.BaseSettings.OperationNameGenerator = new CustomOperationNameGenerator();
```

Now, if you run your generation with `/events/v3/send` as the `kv.Key` and `post` as the `verb`, you should get `PostEventsV3Send`.

**Comment by christianhelle** [comment]
@damianh Thanks for taking the time to report this. This is clearly a bug. The https://api.hubspot.com/api-catalog-public/v1/apis/events/v3/send OpenAPI spec seems to have a lot of details that I haven't tested with. I'll make sure that the code generated from the HubSpot API will build and work with Refiter

**Comment by christianhelle** [comment]
I have implemented the fix so the generated code for the HubSpot Events API now looks something like this:

```csharp
// <auto-generated>
//     This code was generated by Refitter.
// </auto-generated>

using Refit;
using System.Threading.Tasks;
using System.Collections.Generic;

namespace Hubspotevents.Interface
{
    public interface ICustomBehavioralEventsAPI
    {

        /// <summary>
        /// Endpoint to send an instance of a behavioral event
        /// </summary>
        [Post("/events/v3/send")]
        Task PostEventsV3Send([Body] BehavioralEventHttpCompletionRequest body);
    }
}
```

**Comment by christianhelle** [comment]
This issue in resolved in [v0.5.28](https://www.nuget.org/packages/Refitter/0.5.28)

**Comment by damianh** [comment]
Nice, will take it for a spin.
